src/abstract-interpretation/data-frame/mappers/arguments.ts

```diff
@@ -0,0 +1,270 @@
+	let arg = undefined;
+
+	if(name !== undefined) {
+		arg = args.find(arg => resolveIdToArgName(arg, info) === name);
+	}
+	const hasArgPos = arg === undefined && pos >= 0 && pos < args.length && args[pos] !== EmptyArgument && args[pos].name === undefined;
+
+	if(hasArgPos) {
+		arg = args[pos];
+	}
+	return arg;
+}
+
+/**
+ * Get all function arguments of a function call node in the data flow graph.
+ *
+ * @param node - The function call node to get the arguments for
+ * @param dfg  - The data flow graph for retrieving the arguments
+ * @returns The arguments of the function call in the data flow graph
+ */
+export function getFunctionArguments(
+	node: RFunctionCall<ParentInformation>,
+	dfg: DataflowGraph
+): readonly RFunctionArgument<ParentInformation>[] {
+	const vertex = dfg.getVertex(node.info.id);
+
+	if(vertex?.tag === VertexType.FunctionCall && dfg.idMap !== undefined) {
+		const idMap = dfg.idMap;
+
+		return vertex.args
+			.map(arg => arg === EmptyArgument ? arg : idMap.get(arg.nodeId))
+			.map(arg => arg === EmptyArgument || arg?.type === RType.Argument ? arg : toUnnamedArgument(arg, idMap));
+	}
+	return node.arguments;
+}
+
+/**
+ * Gets all nested symbols in an expression that have no outgouing edges in the data flow graph.
+ * @param expression - The expression to get the symbols from
+ * @param dfg        - The data flow graph for checking the outgoing edges
+ * @returns The name of all unresolved symbols in the expression
+ */
+export function getUnresolvedSymbolsInExpression(
+	expression: RNode<ParentInformation> | typeof EmptyArgument | undefined,
+	dfg?: DataflowGraph
+): string[] {
+	if(expression === undefined || expression === EmptyArgument || dfg === undefined) {
+		return [];
+	}
+	switch(expression.type) {
+		case RType.ExpressionList:
+			return [...expression.children.flatMap(child => getUnresolvedSymbolsInExpression(child, dfg))];
+		case RType.FunctionCall:
+			return [...expression.arguments.flatMap(arg => getUnresolvedSymbolsInExpression(arg, dfg))];
+		case RType.UnaryOp:
+			return [...getUnresolvedSymbolsInExpression(expression.operand, dfg)];
+		case RType.BinaryOp:
+			return [...getUnresolvedSymbolsInExpression(expression.lhs, dfg), ...getUnresolvedSymbolsInExpression(expression.rhs, dfg)];
+		case RType.Access:
+			return [...getUnresolvedSymbolsInExpression(expression.accessed, dfg), ...expression.access.flatMap(arg => getUnresolvedSymbolsInExpression(arg, dfg))];
+		case RType.Pipe:
+			return [...getUnresolvedSymbolsInExpression(expression.lhs, dfg), ...getUnresolvedSymbolsInExpression(expression.rhs, dfg)];
+		case RType.Argument:
+			return [...getUnresolvedSymbolsInExpression(expression.value, dfg)];
+		case RType.Symbol:
+			if(isUseVertex(dfg.getVertex(expression.info.id)) && (dfg.outgoingEdges(expression.info.id)?.size ?? 0) === 0) {
+				return [unquoteArgument(expression.content)];
+			} else {
+				return [];
+			}
+		default:
+			return [];
```

Comment:
Is this the reason you do not use one of the visitors? In general I would prefer you use the visitor as you may miss some children etc. when we change the AST structure in the fturue